vsphere/internal/helper/hostsystem/host_system_helper.go

```diff
@@ -51,6 +53,13 @@ func FromID(client *govmomi.Client, id string) (*object.HostSystem, error) {
 	defer cancel()
 	hs, err := finder.ObjectReference(ctx, ref)
 	if err != nil {
+		if soap.IsSoapFault(err) {
```

Comment:
Check out [this](https://github.com/terraform-providers/terraform-provider-vsphere/blob/master/vsphere/internal/helper/viapi/vim_helper.go#L54) function. Should make this check a little easier.

Reply:
Yeah one of my concerns while writing this resource was that I'd probably end up re-implementing things that are already done elsewhere in the provider :).

Thanks for the pointer!